src/adapter/src/rbac.rs

```diff
@@ -590,19 +610,21 @@ fn generate_required_privileges(
             .collect(),
         Plan::ShowCreate(plan) => {
             let item = catalog.get_item(&plan.id);
-            BTreeSet::from([(
+            vec![(
                 item.name().qualifiers.clone().into(),
                 AclMode::USAGE,
                 role_id,
-            )])
+            )]
         }
 
         Plan::Peek(plan) => {
             let mut privileges =
                 generate_read_privileges(catalog, depends_on.iter().cloned(), role_id);
+            // source hasn't been fully optimized yet, so it might actually be a constant,
```

Comment:
Does this mean we might require too many permissions for some queries? Seems fine.

Reply:
Yeah, we will require cluster privileges even if the query doesn't end up using a cluster.